src/ng-select/lib/ng-templates.directive.ts

```diff
@@ -1,80 +1,79 @@
-import { Directive, ElementRef, Input, OnChanges, SimpleChanges, TemplateRef } from '@angular/core';
+import { Directive, ElementRef, inject, OnChanges, SimpleChanges, TemplateRef, input } from '@angular/core';
 import { escapeHTML } from './value-utils';
 
 @Directive({ selector: '[ngItemLabel]' })
 export class NgItemLabelDirective implements OnChanges {
-	@Input() ngItemLabel: string;
-	@Input() escape = true;
-
-	constructor(private element: ElementRef<HTMLElement>) {}
+	ngItemLabel = input<string>();
+	escape = input(true);
+	private element = inject<ElementRef<HTMLElement>>(ElementRef<HTMLElement>);
 
 	ngOnChanges(changes: SimpleChanges) {
-		this.element.nativeElement.innerHTML = this.escape ? escapeHTML(this.ngItemLabel) : this.ngItemLabel;
```

Comment:
@pavankjadda , same here, I think ngOnChanges will never be triggered with Signal Inputs.